test/functional/test_site/expected/index.html

```diff
@@ -557,6 +561,9 @@ <h1 id="test-plugin-in-markbind-plugins">Test plugin in markbind/plugins<a class
           <h1 id="markbind-plugin-pre-render">Markbind Plugin Pre-render<a class="fa fa-anchor" href="#markbind-plugin-pre-render"></a></h1>
           <p>Node Modules Plugin Post-render</p>
         </div>
+        <h1 id="test-search-indexing">Test search indexing<a class="fa fa-anchor" href="#test-search-indexing"></a></h1>
+        <h2 class="noindex" id="level-2-header-inside-headingsearchindex-with-noindex-attribute-should-not-be-indexed">Level 2 header (inside headingSearchIndex) with noindex attribute should not be indexed</h2>
```

Comment:
Should we remove the `index` and `noindex` classes from the final rendered HTML?

Reply:
Thanks for pointing it out! I think that it makes sense to have it omitted in the output HTML, but our current site building process seems to make this difficult because 1) the index is only built **after** the HTML is generated and written to the file system, and 2) the index is built by reading the generated HTML file.

I think it might be possible to overcome this by separating the generation step from the writing step, and inserting the indexing step in between the two, but I feel like this might complicate the build process.

We *could* remove these classes in `setup.js`, but I'm not sure if there is any value to it.

Would love to get inputs from the rest on the team on this! 🙂

Reply:
> We could remove these classes in `setup.js`, but I'm not sure if it's a good idea.

There's no point removing those classes on page load is there? It would still be in the rendered HTML.

Reply:
It would remove those classes from the affected elements after the initial render, but agreed - I don't think there's much value in doing that.